pg: send RowDescription in response to Describe (statement variant)

kostya-sh opened the pull request:
According to https://www.postgresql.org/docs/9.1/static/protocol-flow.html

  The Describe message (statement variant) specifies the name of an
  existing prepared statement (or an empty string for the unnamed
  prepared statement). The response is a ParameterDescription message
  describing the parameters needed by the statement, followed by a
  RowDescription message describing the rows that will be returned when
  the statement is eventually executed (or a NoData message if the
  statement will not return rows)

kostya-sh committed:
pg: send RowDescription in response to Describe (statement variant)

According to https://www.postgresql.org/docs/9.1/static/protocol-flow.html

  The Describe message (statement variant) specifies the name of an
  existing prepared statement (or an empty string for the unnamed
  prepared statement). The response is a ParameterDescription message
  describing the parameters needed by the statement, followed by a
  RowDescription message describing the rows that will be returned when
  the statement is eventually executed (or a NoData message if the
  statement will not return rows)

svladykin commented:
Can we have a test which fails with the old implementation and passes with your fix?

kostya-sh committed:
pg: add test that verifies PGServer response to Describe message

Prior to the fix the test would fail with NPE:

  Exception in thread "main" java.lang.NullPointerException
      at org.postgresql.jdbc2.AbstractJdbc2ResultSet.checkColumnIndex(AbstractJdbc2ResultSet.java:2619)
      at org.postgresql.jdbc2.AbstractJdbc2ResultSet.checkResultSet(AbstractJdbc2ResultSet.java:2637)
      at org.postgresql.jdbc2.AbstractJdbc2ResultSet.getInt(AbstractJdbc2ResultSet.java:2005)
      at org.h2.test.server.TestPGServer.testPrepareWithUnspecifiedType(TestPGServer.java:76)
      at org.h2.test.server.TestPGServer.test(TestPGServer.java:44)
      at org.h2.test.server.TestPGServer.main(TestPGServer.java:29)

kostya-sh commented:
I have discovered the problem while using Go `github.com/lib/pq` driver. 

Triggering the same behaviour using the JDBC driver is a bit trickier but is possible. I have added a test. Please have a look.
